semver/eq.ts

```diff
@@ -1,9 +1,9 @@
 // Copyright 2018-2024 the Deno authors. All rights reserved. MIT license.
-import { compare } from "./compare.ts";
-import type { SemVer } from "./types.ts";
+import { equals } from "./equals.ts";
 
-/** Returns `true` if they're logically equivalent, even if they're not the exact
- * same version object. */
-export function eq(s0: SemVer, s1: SemVer): boolean {
-  return compare(s0, s1) === 0;
-}
+/**
+ * Returns `true` if they're logically equivalent, even if they're not the exact same version object.
+ *
+ * @deprecated (will be removed after 0.213.0) Use {@linkcode equals} instead.
+ */
+export const eq = equals;
```

Comment:
Can you do the deprecation like the below structure? This structure of deprecation has a problem in showing the `@deprecation` note in IDE (#4027).

```ts
/**
 * @deprecated (will be removed after 0.213.0) Use {@linkcode equals} instead.
 *
 * Returns `true` if they're logically equivalent, even if they're not the exact
 * same version object. 
 */
export function eq(s0: SemVer, s1: SemVer): boolean {
  return equals(s0, s1);
}
```

Reply:
done